addon/modifiers/in-viewport.js

```diff
@@ -4,92 +4,97 @@ import { inject as service } from '@ember/service';
 import { DEBUG } from '@glimmer/env';
 import Modifier from 'ember-modifier';
 import deepEqual from 'fast-deep-equal';
+import { registerDestructor } from '@ember/destroyable';
+import { macroCondition, dependencySatisfies } from '@embroider/macros';
 
 const WATCHED_ELEMENTS = DEBUG ? new WeakSet() : undefined;
 
-export default class InViewportModifier extends Modifier {
-  @service inViewport;
-
-  name = 'in-viewport';
-
-  lastOptions;
-
-  get options() {
-    // eslint-disable-next-line no-unused-vars
-    const { onEnter, onExit, ...options } = this.args.named;
-    return options;
-  }
-
-  get hasStaleOptions() {
-    return !deepEqual(this.options, this.lastOptions);
-  }
-
-  validateArguments() {
-    assert(
-      `'{{in-viewport}}' does not accept positional parameters. Specify listeners via 'onEnter' / 'onExit'.`,
-      this.args.positional.length === 0
-    );
-    assert(
-      `'{{in-viewport}}' either expects 'onEnter', 'onExit' or both to be present.`,
-      typeof this.args.named.onEnter === 'function' ||
-        typeof this.args.named.onExit === 'function'
-    );
-  }
-
-  @action
-  onEnter(...args) {
-    if (this.args.named.onEnter) {
-      this.args.named.onEnter.call(null, this.element, ...args);
+let modifier;
+
+if (macroCondition(dependencySatisfies('ember-modifier', '>=3.2.0 || 4.x'))) {
```

Comment:
@And012 now it's missing `else` branch for compatibility with versions that provide old API